pkg/util/etcdutil/migrate.go

```diff
@@ -91,72 +89,29 @@ func waitEtcdTPRReady(restClient restclient.Interface, interval, timeout time.Du
 	return nil
 }
 
-func getBootEtcdPodIP(kubecli kubernetes.Interface) (string, error) {
-	var ip string
-	err := wait.Poll(5*time.Second, 60*time.Second, func() (bool, error) {
-		podList, err := kubecli.CoreV1().Pods(api.NamespaceSystem).List(v1.ListOptions{
-			LabelSelector: "k8s-app=boot-etcd",
-		})
-		if err != nil {
-			glog.Errorf("failed to list 'boot-etcd' pod: %v", err)
-			return false, err
-		}
-		if len(podList.Items) < 1 {
-			glog.Warningf("no 'boot-etcd' pod found, retrying after 5s...")
-			return false, nil
-		}
-
-		pod := podList.Items[0]
-		ip = pod.Status.PodIP
-		if len(ip) == 0 {
-			return false, nil
-		}
-		return true, nil
-	})
-	return ip, err
+func createBootstrapEtcdService(restclient restclient.Interface, svcPath string) error {
+	svc, err := ioutil.ReadFile(svcPath)
+	if err != nil {
+		return err
+	}
+	return restclient.Post().RequestURI("/api/v1/namespaces/kube-system/services").SetHeader("Content-Type", "application/json").Body(svc).Do().Error()
```

Comment:
nit: We use the restclient for the TPR because the normal client isn't aware of the underlying type - but for services we could use something like `client.Core().Services().Update(service)`. Don't really need to change this, however. v1/service rest endpoint is not going anywhere.